gef-ui/server/controllers/about-facility/index.validate-and-update-about-facility.test.js

```diff
@@ -826,5 +806,63 @@ describe('validateAndUpdateAboutFacility', () => {
         }),
       );
     });
+
+    it('redirects user to facility end date page if using facility end date', async () => {
+      mockRequest.body.facilityType = CONSTANTS.FACILITY_TYPE.CASH;
```

Comment:
just fwiw id prefer immer here (or cloning at a stretch) as this is modifying the original object (I'm not sure how that carries through with jest tests without testing it myself)

Reply:
Yeah this whole pattern is an ick but its done intentionally in the beforeEach
```
  beforeEach(() => {
    mockResponse = MockResponse();
    mockRequest = MockRequest();
    mockFacilityResponse = MockFacilityResponse();

    api.getFacility.mockResolvedValue(mockFacilityResponse);
    api.updateFacility.mockResolvedValue({});
    api.updateApplication = updateApplicationSpy;
  });
```